### PR TITLE
junit test code for hashcode and equals method

### PR DESCRIPTION
**Pull request by ykgautam**

### CODE REVIEW

File: java-examples/src/main/java/edu/practice/hashequals/Human.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package edu.practice.hashequals;
+
+public class Human {
+	
+	int adharNumber ;
```
Comment (Ganeshgautam) [inline java-examples/src/main/java/edu/practice/hashequals/Human.java:5]: why not private?
```suggestion
	private int adharNumber ;
```

File: java-examples/src/main/java/edu/practice/hashequals/Human.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	String name;
```
Comment (Ganeshgautam) [inline java-examples/src/main/java/edu/practice/hashequals/Human.java]:
```suggestion
	private String name;
```

File: java-examples/src/main/java/edu/practice/hashequals/Human.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	@Override
+	public int hashCode() {
+		return adharNumber;
+	}
+	@Override
+	public boolean equals(Object obj) {
+		Human h = (Human) obj;
+		if(this == obj) // if yash.equals(yash) 
+		{
+//			System.out.println("fff");
```
Comment (Ganeshgautam) [inline java-examples/src/main/java/edu/practice/hashequals/Human.java:22]: remove all commented out code.

File: java-examples/src/test/java/edu/practice/hashequals/HumanTest.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
+package edu.practice.hashequals;
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.jupiter.api.Test;
+
+class HumanTest {	
+	static private int adharNo1 = 111;
```
Comment (Ganeshgautam) [inline java-examples/src/test/java/edu/practice/hashequals/HumanTest.java:8]: why static? 
also you should keep following order if there is some requirement for static:
`private static SomeType someField;`

File: java-examples/src/test/java/edu/practice/hashequals/HumanTest.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+	static private String name1 = "yash";
+	
+	static int adharNo2 = 111;
+	static String name2 = "yash";
```
Comment (Ganeshgautam) [inline java-examples/src/test/java/edu/practice/hashequals/HumanTest.java:12]: remove static from all these fields.

File: java-examples/src/test/java/edu/practice/hashequals/HumanTest.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+	@Test
+	void testHashcode() {
```
Comment (Ganeshgautam) [inline java-examples/src/test/java/edu/practice/hashequals/HumanTest.java:14]: Add more cases for null cases, empty name and with human2 as null.

**Review by Ganeshgautam — CHANGES_REQUESTED** [review]

**Comment by ykgautam** [comment]
If I make adharNumber as private den I can't get that number in test
class.. scope visibility kam ho jati
 Protected bana k chal jata hai

**Comment by Ganeshgautam** [comment]
> If I make adharNumber as private den I can't get that number in test class.. scope visibility kam ho jati Protected bana k chal jata hai
> […](#)

what happened to abstraction? 
Expose a getter and make the fields private.

**Comment by ykgautam** [comment]
okay.. i will do this creating getter method for it

**Pull request merged**